languages/python/exercises/concept/tuples/tuples_test.py

```diff
@@ -0,0 +1,207 @@
+                                (('Carved Wooden Elephant', '8C'), ('Abandoned Lighthouse', ('4', 'B') ,'Blue'))
+                            ]
+        result_data = [True, True, True, True, True, False, False, False, False, False]
+
+        for item, result in zip(input_data, result_data):
+            with self.subTest("do the coordinates match once reformatted?", item=item, result=result):
+                self.assertEqual(compare_records(item[0], item[1]), result)
+
+    def test_create_record(self):
+        input_data=[
+                            (('Angry Monkey Figurine', '5B'), ('Stormy Breakwater', ('5', 'B') ,'Purple')),
+                            (('Carved Wooden Elephant', '8C'), ('Foggy Seacave', ('8', 'C'), 'Purple')),
+                            (('Amethyst  Octopus', '1F'), ('Aqua Lagoon (Island of Mystery)', ('1', 'F') ,'Yellow')),
+                            (('Antique Glass Fishnet Float', '3D'), ('Spiky Rocks', ('3', 'D'),'Yellow')),
+                            (('Silver Seahorse', '4E'), ('Hidden Spring (Island of Mystery)', ('4', 'E') ,'Yellow')),
+                            (('Amethyst  Octopus', '1F'), ('Seaside Cottages', ('1', 'C') ,'Blue')),
+                            (('Angry Monkey Figurine', '5B'), ('Aqua Lagoon (Island of Mystery)', ('1', 'F') ,'Yellow')),
+                            (('Antique Glass Fishnet Float', '3D'), ('Deserted Docks', ('2', 'A') ,'Blue')),
+                            (('Brass Spyglass', '4B'), ('Spiky Rocks', ('3', 'D'),'Yellow')),
+                            (('Carved Wooden Elephant', '8C'), ('Abandoned Lighthouse', ('4', 'B') ,'Blue'))
+                            ]
+        result_data = [
+                                ('Angry Monkey Figurine', '5B', 'Stormy Breakwater', ('5', 'B'), 'Purple'),
+                                ('Carved Wooden Elephant', '8C', 'Foggy Seacave', ('8', 'C'), 'Purple'),
+                                ('Amethyst  Octopus', '1F', 'Aqua Lagoon (Island of Mystery)', ('1', 'F'), 'Yellow'),
+                                ('Antique Glass Fishnet Float', '3D', 'Spiky Rocks', ('3', 'D'), 'Yellow'),
+                                ('Silver Seahorse', '4E', 'Hidden Spring (Island of Mystery)', ('4', 'E'), 'Yellow'),
+                                'not a match',
+                                'not a match',
+                                'not a match',
+                                'not a match',
+                                'not a match'
+                            ]
+
+        for item, result in zip(input_data, result_data):
+            with self.subTest("make record if coordinates match", item=item, result=result):
+                self.assertEqual(create_record(item[0], item[1]), result)
+
+    def test_clean_up(self):
+
+        input_data = (
+                                ("Scrimshaw Whale's Tooth", '2A', 'Deserted Docks', ('2', 'A'), 'Blue'),
```

Comment:
Minor nitpick: there's inconsistency between this line (which uses double quotes for this string literal) and other lines (which use single quotes, and escape the embedded apostrophe). I recommend making the others match this to avoid escaping quotes.

Reply:
Yup.  _Thank you_ for catching that.  The quotes were a nightmare when it came to checkout `std.out`.  Next time, I won't name something with an apostrophe int it!  😆   Fixed in latest push.

Reply:
I find running [`black`](https://github.com/psf/black) on my code to be useful. Among other minor style changes, it would've converted all strings to use double quotes, unless they contain a double quote, then it uses single quotes.

Reply:
Yeah.  Somehow, in my long twisty path I've developed a habit of single quotes.  It may be time to remedy that .. and to set up Black to work with Pycharm.

Reply:
In VS Code, I have `flake8` run on file save and alert me of violations, then I manually run `black` to fix the violations it can. I don't like editors modifying my files implicitly.

I don't use PyCharm, but I'm sure there's a way to set up something similar.